File: packages/flutter/lib/src/exceptions.dart
/// Exceptions thrown by the TBD Agents SDK.
library;

/// Base class for all TBD Agents SDK exceptions.
class TbdAgentsException implements Exception {
  const TbdAgentsException(this.message);

  final String message;

  @override
  String toString() => 'TbdAgentsException: $message';
}

/// Thrown when the underlying HTTP transport fails (network error, timeout, etc.).
class TransportException extends TbdAgentsException {
  const TransportException(super.message, {this.cause});

  final Object? cause;

  @override
  String toString() =>
      cause == null ? 'TransportException: $message' : 'TransportException: $message (caused by $cause)';
}

/// Thrown when the API returns a non-2xx response.
class ApiException extends TbdAgentsException {
  const ApiException(
    super.message, {
    required this.statusCode,
    this.body,
  });

  /// HTTP status code returned by the server.
  final int statusCode;

  /// Parsed response body (may be a [Map], [List], [String], or `null`).
  final Object? body;

  @override
  String toString() => 'ApiException($statusCode): $message';
}

File: packages/python/src/tbd_agents/exceptions.py
from __future__ import annotations

from typing import Any

import httpx


class TbdAgentsError(Exception):
    """Base SDK exception."""


class TransportError(TbdAgentsError):
    """Raised when the underlying HTTP transport fails."""

    def __init__(self, message: str, *, cause: Exception | None = None) -> None:
        super().__init__(message)
        self.cause = cause


class ApiError(TbdAgentsError):
    """Raised for non-successful API responses."""

    def __init__(
        self,
        message: str,
        *,
        status_code: int,
        response: httpx.Response,
        body: Any | None = None,
    ) -> None:
        super().__init__(message)
        self.status_code = status_code
        self.response = response
        self.body = body

